base-ledger-keyring.js

```diff
@@ -0,0 +1,619 @@
+const { EventEmitter } = require('events')
+const HDKey = require('hdkey')
+const ethUtil = require('ethereumjs-util')
+const sigUtil = require('eth-sig-util')
+const { TransactionFactory } = require('@ethereumjs/tx')
+
+const pathBase = 'm'
+const hdPathString = `${pathBase}/44'/60'/0'`
+const type = 'Ledger Hardware'
+
+const BRIDGE_URL = 'https://metamask.github.io/eth-ledger-bridge-keyring'
+
+const MAX_INDEX = 1000
+const NETWORK_API_URLS = {
+  ropsten: 'http://api-ropsten.etherscan.io',
+  kovan: 'http://api-kovan.etherscan.io',
+  rinkeby: 'https://api-rinkeby.etherscan.io',
+  mainnet: 'https://api.etherscan.io',
+}
+
+class BaseLedgerKeyring extends EventEmitter {
+  constructor (opts = {}) {
+    super()
+    this.accountDetails = {}
+    this.bridgeUrl = null
```

Comment:
This `bridgeUrl` is specific to the MV2 implementation. But it would be difficult to move it from here given that it's involved with the serialize/deserialize process.